### DISCUSSION TITLE
IP address from where message opened is my Postal instance IP address

### DISCUSSION THREAD

**Discussion by Ascar699** [postalserver/postal#2179] [Help with using Postal]
Hello, everyone.

I moved Postal from behind HAProxy and assigned a public IP address to the network interface.  Postal works fine, however all message opens and clicks show as originating from my own IP address.

My Caddyfile parameter:

# Update this to match the hostname you wish to use for accessing the
# Postal web interface.
click.xxxxx-me.com {
  reverse_proxy 212.xxx.xxx.75:5000
}

# If you use open/click tracking you may need to add additional hosts
# to this file using the format below.
#
click.xxxxxx-me.com {
   reverse_proxy 212.xxx.xxx.75:5000:5000 {
     header_up X-Postal-Track-Host "1"
   }
 }

Postal is also bound to the same public IP address.  Clicks and opens work but what I see in Postal GUI is this:

`Message Viewed

Opened from 212.xxx.xxx.75 (Mozilla/5.0 (Linux; Android 10; Redmi Note 8 Pro Build/QP1A.190711.020; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/105.0.5195.136 Mobile Safari/537.36)`

Can't crack it myself.  Please suggest something.  Thanks

**Comment by willpower232** [comment]
I believe Caddy is using the X-Forwarded-For header correctly by default so I'd presume that your HAProxy needs to send through the header it receives without modifying it.

**Comment by Ascar699** [comment]
No, Bill, we discussed this before, so I moved the server from behind the HAProxy and exposed it directly to the internet.  There is no any reverse proxy right now.  This is what makes me scratch my head.

**Reply by willpower232** [reply]
ah my mistake, then you should be giving Caddy 127.0.0.1 rather than the 212... address

**Reply by Ascar699** [reply]
Ah, somehow missed your message.  Let me change the IP address to the localhost and re-test.
Thanks for you for the suggestion

**Reply by Ascar699** [reply]
Ooops, I am on already on 127.0.0.1:5000 and I am still getting the message reads from the IP address of my Postal instance.

**Reply by Ascar699** [reply]
Tried changing the reverse_proxy to public IP address and keeping the localhost address for the tracking host.  Same result :(

`Update this to match the hostname you wish to use for accessing the
Postal web interface.
smtp.xxxx-me.com {
  reverse_proxy 212.xxx.xx.75:5000
}

If you use open/click tracking you may need to add additional hosts
to this file using the format below.

click.xxxxxx-me.com {
   reverse_proxy 127.0.0.1:5000 {
     header_up X-Postal-Track-Host "1"
   }
 }`
